contract_sale_generation/models/account_analytic_account.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2004-2010 OpenERP SA
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# Copyright 2015 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2017 Pesol (<http://pesol.es>)
+# Copyright 2017 Angel Moya <angel.moya@pesol.es>
+# Copyright 2018 Therp BV <https://therp.nl>.
+# Copyright 2019 Sylvain Van Hoof <sylvain@okia.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.model
+    def _prepare_sale_line(self, line, order_id):
+        sale_line = self.env['sale.order.line'].new({
+            'order_id': order_id,
+            'product_id': line.product_id.id,
+            'product_qty': line.quantity,
+            'product_uom_qty': line.quantity,
+            'product_uom': line.uom_id.id,
+        })
+        # Get other sale line values from product onchange
+        sale_line.product_id_change()
+        sale_line_vals = sale_line._convert_to_write(sale_line._cache)
+        # Insert markers
+        contract = line.analytic_account_id
+        lang_obj = self.env['res.lang']
+        lang = lang_obj.search(
+            [('code', '=', contract.partner_id.lang)])
+        date_format = lang.date_format or '%m/%d/%Y'
+        name = self._insert_markers(line, date_format)
+        sale_line_vals.update({
+            'name': name,
+            'discount': line.discount,
+            'price_unit': line.price_unit,
```

Comment:
In this part we're missing UOM . Is it possible to add this part https://github.com/OCA/contract/pull/284/files